### PR TITLE
test(utils): add unit test for `transform_path`

### PR DESCRIPTION
**Pull request by jamestrew**
Adds os-agnostic unit tests for `transform_path` testing and fixes a couple of general edge cases and Windows specific path separator related bugs.

I was able to run these tests against my CI changes in https://github.com/nvim-telescope/telescope.nvim/pull/3011 to confirm they pass for all OSs.

### CODE REVIEW

File: lua/telescope/utils.lua [hunk 1]
```diff
@@ -53,7 +53,7 @@ utils.path_expand = function(path)
 
   path = path:gsub("%$([%w_]+)", vim.loop.os_getenv)
   path = path:gsub("/+", "/")
-  if iswin then
+  if utils.iswin then
```
Comment (jamestrew) [inline lua/telescope/utils.lua:56]: convert all `/` into a `\`?
at least this is closer to `vim.fn.expand`.
